rclpy/test/test_handle.py

```diff
@@ -47,52 +47,3 @@ def test_handle_destroyed_when_not_used():
                 pass
     finally:
         rclpy.shutdown(context=context)
-
-
-def test_handle_destroys_dependents():
-    context = rclpy.context.Context()
-    rclpy.init(context=context)
-
-    try:
-        node1 = rclpy.create_node('test_handle_destroys_dependents1', context=context)
-        node2 = rclpy.create_node('test_handle_destroys_dependents2', context=context)
-        node2.handle.requires(node1.handle)
-
-        with node1.handle:
-            pass
-        with node2.handle:
-            pass
-
-        node1.handle.destroy()
-        with pytest.raises(InvalidHandle):
-            with node1.handle:
-                pass
-        with pytest.raises(InvalidHandle):
-            with node2.handle:
-                pass
-    finally:
-        rclpy.shutdown(context=context)
-
-
-def test_handle_does_not_destroy_requirements():
-    context = rclpy.context.Context()
-    rclpy.init(context=context)
-
-    try:
-        node1 = rclpy.create_node('test_handle_does_not_destroy_requirements1', context=context)
-        node2 = rclpy.create_node('test_handle_does_not_destroy_requirements2', context=context)
-        node2.handle.requires(node1.handle)
-
-        with node1.handle:
-            pass
-        with node2.handle:
-            pass
-
-        node2.handle.destroy()
-        with pytest.raises(InvalidHandle):
-            with node2.handle:
-                pass
-        with node1.handle:
-            pass
-    finally:
-        rclpy.shutdown(context=context)
```

Comment:
@ivanpauno why were these tests dropped? they certainly have to be rewritten but still apply conceptually, no?

Reply:
Those tests were relaying in using `requires`, which is not exposed more.
I think that the tests I wrote in https://github.com/ros2/rclpy/blob/39a93ac5cde247b53137441e34f431d68ba211c4/rclpy/src/test/rclpy_common/test_handle.cpp are a good replacement.

Reply:
I see, fair enough.